firedrake/tsfc_interface.py

```diff
@@ -90,27 +90,29 @@ def _cache_store(cls, key, val):
         comm.barrier()
 
     @classmethod
-    def _cache_key(cls, form, name, parameters, number_map):
+    def _cache_key(cls, form, name, parameters, number_map, interface):
         # FIXME Making the COFFEE parameters part of the cache key causes
         # unnecessary repeated calls to TSFC when actually only the kernel code
         # needs to be regenerated
         return md5((form.signature() + name
                     + str(sorted(default_parameters["coffee"].items()))
                     + str(sorted(parameters.items()))
-                    + str(number_map)).encode()).hexdigest(), form.ufl_domains()[0].comm
+                    + str(number_map)
+                    + str(type(interface))).encode()).hexdigest(), form.ufl_domains()[0].comm
```

Comment:
`type(interface)` is a fully-qualified name, so this must be unique, so that is OK I think